src/client/datascience/jupyter/kernels/kernelExecution.ts

```diff
@@ -168,12 +166,12 @@ export class KernelExecution implements IDisposable {
         }
     }
 
-    public cancelAllCells(document: NotebookDocument): void {
+    public async cancelAllCells(document: NotebookDocument): Promise<void> {
         if (this.documentExecutions.get(document)) {
             this.documentExecutions.get(document)!.cancel();
         }
         traceInfo('Cancel document execution');
-        document.cells.forEach((cell) => this.cancelCell(cell));
+        await Promise.all(document.cells.map((cell) => this.cancelCell(cell)));
```

Comment:
Floating promises fixed, we are waiting when cancelling a single cell, now when cancelling all cell, we were not waiting.